core/controllers/admin.py

```diff
@@ -854,3 +854,26 @@ def post(self):
             self.render_json({})
         else:
             raise self.InvalidInputException('This app cannot send emails.')
+
+
+class UpdateUsernameHandler(base.BaseHandler):
+    """Handler for renaming usernames."""
+
+    @acl_decorators.can_access_admin_page
+    def put(self):
+        current_username = self.payload.get('current_username', None)
+        new_username = self.payload.get('new_username', None)
+        if new_username is not None:
+            if len(new_username) > 30:
```

Comment:
Change 30 to be MAX_USERNAME_LENGTH and put it as a constant in the appropriate place.

Reply:
I've added a constant in this file itself. There exists a constant in the user_services with the same name as well but I'm not changing that at the moment since that would apply to the new users that visit the signup page as well and we don't want that right now. That change would be done via #9051 which will be merged after the long usernames are changed manually via this PR.

Reply:
Actually, if you could apply this check to new usernames, that would be better, so that we don't run into the same problem with new usernames going forward. The only place where you still need to keep the old length checks are places that handle existing models -- once the existing names are migrated, we can drop that special case. But the first step in a migration is always to prevent new "bad" stuff from coming in.

Reply:
Done. I've added the changes that were supposed to be done after this PR gets merged but I see no harm in making these changes simultaneously either.